sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140046&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140046&tab=artifacts#/) on master @ [365b4da8bd02c06ee59d2130a56dec74ffc9ce21](https://github.com/cockroachdb/cockroach/commits/365b4da8bd02c06ee59d2130a56dec74ffc9ce21):


```
            • sort
            │ order: +k
            │
            └── • filter
                │ filter: st_intersects('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFP2zwUhu-_X2GdG1p9prWdlA1fFUbYMpWWpZ02NFcoa85YRLEz2506of73KQ0DAmogvnB7jv2ck_e17Ftwv5YgIfp6PjqKx6RzEk9n00-jLplGo-jdjFyT02RyRq7QXPr0-xLJlw9REhHnL3Pt0TpceNfZO_s8msXnk3g863RETxDRE11KOkGPkaDHut09Kd9Hk7NollzQstZNl0ySkyghxxfkGihok-E4vUEH8htwoCBgTqGwZoHOGVumb7eb4mwNklHIdbHyZXpOYWEsgrwFn_slgoSx2TdFfwAUMvRpvtxu21AwK_8AOZ9eIciDDX1UmDcXnpUGJJhmaPusVh7u_RmW6i5zneEaKEyLVDtJ-gqOlVr_yJRac6bUmr00wX5bhisgqc5IwIjxP9E62KWZt9Ec699oPWan-dKjRdvndeH_1qN1YYnRZMglcaVq4nxqvdyqCN4MlGKCKcXYSxMQ1FlbrBT_RD2FycpLMuQ7fRBtfPhocn139GLX0Rc2v0ntn4fWdCh2dg_adL93P6j3rvLy6XVknIWsGuLulzNe_Tk8OrwfjIfP4oedtfE2fBbvyce3eii6r_A8bKN6aqxH2w_rmof8_53lB7XyL7wTCbrCaIeveijYZk4Bsyus3iJnVnaB59Ystm2qcLLltokMna9WD6og1tVS-YGPYd4Ii2ZYNMJBMxw0wmEzHDbCg2Z40AizJ_B889_fAAAA___nMRL3
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • sort
            │ order: +k
            │
            └── • filter
                │ filter: st_intersects('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFP2zAQx9_3Kax7odW81naCNvxUGGHLVChLO21orlDW3FhEsTPbnTKhfvcpCQNa1ED84PbO_v3P_7OcW3C_lyAh-nY-PozPSO84ns6mn8d9Mo3G0fsZuSYnyeSUXKG59OmPJZKvH6MkIs5f5tqjdbjwrrd3-mU8i88n8dms1xMDQcRA9CnpBQNGggHr9_ek_BBNTqNZckErrZs-mSTHUUKOLsg1UNAmw7P0Bh3I78BhTqGwZoHOGVulbusNcVaCZBRyXax8lZ5TWBiLIG_B536JIGFWnTHBNEM7ZEAhQ5_my1r23sKoOsBlrjMsgcK0SLWTZKjgSKnyZ6ZUyZlSJXtugjddGa6ApDojASPG_0LrYL6mYFb-wY_z6RWC5Gv6cs-x_oPWY3aSLz1atEO-afz_elQWlhhNRlwSV7kmzqfWy9pF8HZfKSaYUow9NwFBnXXFKvNb7ilMVl6SEd_ZB9GlD59Mru-uXuy6-sLmN6n9-1CajsTO6kGX6vfdDzZrN3m5_WIYZyFrhrj75Yw3fw4OD-4H4-GT-GHnxngXPon35OOHNxL9F_Q87OJ6aqxHOww3PY_4653y-13kE3SF0Q435Hcps_WcAmZX2HwynFnZBZ5bs6jLNOGk5upEhs43q7wJYt0sVQd8DPNWWLTDohUO2uGgFQ7b4bAV3t-C5-tX_wIAAP__VL3gzA==
--- done: testdata/inverted_filter_geospatial_dist with config 5node: 13 tests, 1 failures
    logic.go:3370: 
        testdata/inverted_filter_geospatial_dist:206: error while processing
    logic.go:3370: testdata/inverted_filter_geospatial_dist:206: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (11.11s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4141630&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4141630&tab=artifacts#/) on master @ [365b4da8bd02c06ee59d2130a56dec74ffc9ce21](https://github.com/cockroachdb/cockroach/commits/365b4da8bd02c06ee59d2130a56dec74ffc9ce21):


```
            vectorized: true
            ·
            • filter
            │ filter: 'BOX(3 3,3 3)' && geom
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk9Fv2j4Qx99_f8XpXmglV9gJ_Db5ibGmGhODDpCGNKMqwzcWDezMNlMmxP8-Jdna0o5Q_HDS3flz5-9Zt0P_Y40Sk_nt8M1gBBfXg-ls-nF4CdNkmLydwXe4mYw_wIrsXUi_rAk-vUsmCbS44II_P687z_yWlP3xPLoGteU8-r-2ZcENMjRW0yjdkEf5GQUyjHDBMHd2Sd5bV4Z31aWBLlByhpnJt6EMLxgurSOUOwxZWBNKHNkrm7c7yFBTSLN1dW3P0G7DA-RDuiKU3T17VFg0F56VyieUanJtflAe7wfTKxXdZUZTgQyneWq8hLbCvlLFV61UIbhSBT9l8OpcRiiE1GiIOdjwjZzHY5rFOZoH5ie5QPomWwdy5NriUPjffFLkDqyBnpDgS9XgQ-qCrFTEr7pK8Ygrxfkpg0BGn4uV4p-oZzjeBgk9cXQO0TlzeG8z8-fro2Nfn7tsk7pfD61ZLzraPT6n-_3048PedVxCqz-eX8QQsxjiy5b856r1ohfMpHPwqhOLNiGfW-PpRZvG9wuGpFdUL7O3W7ekW2eXVZvaHVdcFdDkQ53t1s7A1KnygY9h0QhHzXDUCMfNcNwId5rhTiPMn8CL_X-_AwAA__8rRsTS
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • filter
            │ filter: 'BOX(3 3,3 3)' && geom
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk99v0zAQx9_5K0730k0yqh2XH_JTKctEUVlHW4lKuJpCfZSI1g62i4Kq_u8oCWx0o92SB0t3l89973uWdxh-rFFhOr8evRlewdnFcDqbfhydwzQdpW9n8B0uJ-MPsCJ3E7Mva4JP79JJCh0uuOAPv9e9B3FHqcF4nlyA3nKevGzOquEGGVpn6CrbUED1GQUuGBbeLSkE56vUrv5haEpUnGFui22s0guGS-cJ1Q5jHteECmfVcBPKDPkuR4aGYpav67a3s_cr0ZvcGiqR4bTIbFDQ1TjQuvxqtC4F17rkjx34vC0jNEJmDUgOLn4jH3CxZ-i28c5PiNmKUIk9e7rnof1JPpK5zNeRPPmuODT-t56WhQdnoS8UhMo1hJj5qGoX8tULrXnCteb8sQOBrGmLVebvuWc43kYFfXF0D0mbPbx3uf1z9cmxqy98vsn8rztp1k-Oqss26rfbl4faTV5BZzCen0mQTII876j_voZ-8oSd9NpMNaFQOBvoYKZjnfl-wZDMipo3F9zWL-nau2Ut04TjmqsThkJsqqIJhrYpVQP-C4uTcHIaTk7C8jQsT8K9e_Bi_-x3AAAA__81JZK7
--- done: testdata/inverted_filter_geospatial_dist with config 5node: 17 tests, 1 failures
    logic.go:3370: 
        testdata/inverted_filter_geospatial_dist:286: error while processing
    logic.go:3370: testdata/inverted_filter_geospatial_dist:286: too many errors encountered, skipping the rest of the input
--- done: testdata/inverted_filter_json_array_dist with config 5node: 30 tests, 0 failures
--- done: testdata/explain_analyze_plans with config 5node: 20 tests, 0 failures
--- total progress: 3525 statements/queries
--- total: 3525 tests, 0 failures
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (2.86s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149016&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149016&tab=artifacts#/) on master @ [5aefc070d1e7f5bfd860eae24dea2e7726dc6d8c](https://github.com/cockroachdb/cockroach/commits/5aefc070d1e7f5bfd860eae24dea2e7726dc6d8c):


```
            • sort
            │ order: +k
            │
            └── • filter
                │ filter: st_coveredby('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFP2zAQx9_3Kax7odVMazspG34qjLBlKi1LO21orlCobyyixJntoiLU7z6lYUBBDcQPbu_s3__yP8u-A_d3DhKin6eDg3hIWkfxeDL-NmiTcTSIPk3IFTlORifkEs25Ty_mSH58iZKIOH8-MzdoUV_ctnZOvg8m8ekoHk5aLdERRHREm5JW0GEk6LB2e0fKz9HoJJokZ7SUum6TUXIUJeTwjFwBhdxoHKbX6ED-Ag4UBEwpFNbM0Dljy_TdelOslyAZhSwvFr5MTynMjEWQd-AzP0eQMDS7puj2gIJGn2bz9bYVBbPwj5Dz6SWC3FvRJ8K8XnhS-k8w1Wi7bEMeHtrTL92dZ7nGJVAYF2nuJOkqOFRq-VsrteRMqSV7bYLdpgxXQNJck4AR4_-gdbDNM2_iOc5v0HrUx9nco0Xb5ZvG_69Hy8ISk5M-l8SVronzqfVy7SL40FOKCaYUY69NQDDXTbHS_DP3FEYLL0mfb-2DaNKHrybL749ebDv6wmbXqb19LE37Ymv1oEn1h-4Hm7WrvHx2GxlnIauGuP_ljFd_9g_2Hwbj4Yv4cefG-Bi-iHfk00vdF-03tDxsYnpsrEfbDTct9_n7rfK9DflXnokEXWFyh296J9hqSgH1JVZPkTMLO8NTa2brMlU4WnPrhEbnq9W9Kojzaqn8wKcwr4VFPSxq4aAeDmrhsB4Oa-FePdyrhdkzeLp69y8AAP__-8wR9Q==
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • sort
            │ order: +k
            │
            └── • filter
                │ filter: st_coveredby('01040000000200000001010000009A999999999901409A99999999990140010100000000000000000008400000000000000840', geom)
                │
                └── • index join
                    │ table: geo_table@geo_table_pkey
                    │
                    └── • inverted filter
                        │ inverted column: geom_inverted_key
                        │ num spans: 31
                        │
                        └── • scan
                              missing stats
                              table: geo_table@geom_index
                              spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFP2zwUhu-_X2GdG1p9Xms7QRu-KoywZSqUpZ02NFco1GcsosSZ7aIg1P8-JWFAQQ3EF27PsZ_3-D2WcwfuzxIkRD9Ox_vxCekdxtPZ9Ou4T6bROPo4I1fkKJkck0s05z69WCL5_jlKIuL8-cLcoEV9cdvbOf42nsWnk_hk1uuJgSBiIPqU9IIBI8GA9fs7Un6KJsfRLDmjldR1n0ySwyghB2fkCijkRuNJeo0O5E_gMKdQWLNA54ytUnf1hliXIBmFLC9WvkrPKSyMRZB34DO_RJAwq46YYKrRDhlQ0OjTbFnLPjgYVQc4z3KNJVCYFmnuJBkqOFCq_KWVKjlTqmSvTfCuK8MVkDTXJGDE-N9oHczXFMzKP_pxPr1EkHxN3-45zm_QetRH2dKjRTvkm8b_rUdlYYnJyYhL4irXxPnUelm7CN7vKsUEU4qx1yYgmOuuWGX-mXsKk5WXZMS39kF06cMXk-X3Vy-2XX1hs-vU3j6WpiOxtXrQpfpD94PN2k1ePnswjLOQNUPc_3LGmz97-3sPg_HwRfy4c2N8CF_EO_LpuxuJ_htaHnYxPTXWox2Gm5ZH_P-t8rtd5BN0hckdbshvU2brOQXUl9h8MZxZ2QWeWrOoyzThpObqhEbnm1XeBHHeLFUHfArzVli0w6IVDtrhoBUO2-GwFd59Bs_X__0NAAD___MV38o=
--- done: testdata/inverted_filter_geospatial_dist with config 5node: 14 tests, 1 failures
    logic.go:3370: 
        testdata/inverted_filter_geospatial_dist:234: error while processing
    logic.go:3370: testdata/inverted_filter_geospatial_dist:234: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (9.01s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4150954&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4150954&tab=artifacts#/) on master @ [d1eafdf56b42ca09a169c199916917a144c2c412](https://github.com/cockroachdb/cockroach/commits/d1eafdf56b42ca09a169c199916917a144c2c412):


```
        testdata/inverted_filter_geospatial_dist:238: EXPLAIN (DISTSQL)
        SELECT k FROM geo_table WHERE geom && 'POINT(3.0 3.0)'::geometry
        expected:
            distribution: full
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk29v0z4Qx5__XoV1T7pJ3mon7Q_kR-VPBkFdW9pKgHA1hfooEa0dbAcFVX3vKAlsS6emxA9Oujt_7vw96_bgfmxBQPRxNn4RT8jF63ixXLwfX5JFNI5eLcl3cjOf3pINmjuffNki-fA2mkelvyMyZyz4v7akN5vGk-VFeM1IeM0ue0K8iaa30XL-CShoo3CS7NCB-AwcKASwopBZs0bnjC3D--pSrAoQjEKqs9yX4RWFtbEIYg8-9VsEARNzZbL-ACgo9Em6ra4dKJjcP0DOJxsEMTzQR4V5e-FlKXCOiULbZ43ycK9_VCq_S7XCAigsskQ7QfoSXkpZfFVSFpxJWbBzBq66MlwCSbQiISPGf0Pr4JRm3kVzrH-i9ahu0q1Hi7bPm8L_5qMis8RoMuKCuFI1cT6xXlQqwmdDKVnApGTsnAGCWnXFSvFH6ilMcy_IiJ-cQ9BlDu9Mqv98fXDq6zOb7hL766E1HQUnu4ddut9PP2z2ruOCjIKjZWOccfb0PB888XuisYnnhjZoPPvMJs7RZUY7_KdVZIcVBVQbrLfdmdyucWbNumpTu9OKqwIKna-zw9qJdZ0qH_gY5q1w0A4HrXDYDoet8KAdHrTC7AheHf77HQAA__-eq8eX
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk02P0zAQhu_8itFcuiuZrZ2UD_lUPrIQ1G1LWwkQrlahHkpEawfbRUFV_ztKAru0q7YkB0szk8fzvmPNFv2PFUpMPo4HL9IhXLxOp7Pp-8ElTJNB8moG3-F6MrqBJdnbkH1ZEXx4m0ySKl6D2nAePW1O6IxH6XB2EV9xiK_4ZUfKN8noJplNPiFDYzUNszV5lJ9R4Jxh4eyCvLeuSm3rH1JdouQMc1NsQpWeM1xYRyi3GPKwIpQ4qzRMKNPkuhwZagpZvqqvvZPYr8Td5kZTiQynRWa8hK7Cl0qVX7VSpeBKlfzcgY_bMkIhZEZDzMGGb-Q8zncM7Sbc-_EhWxJKsWP_7zk1P8kF0tf5KpAj1xX7xv_Wk7JwYA30hQRfuQYfMhdk7SJ-9kQpHnGlOD93IJDRbbHK_IF7hqNNkNAXR-cQtZnDO5ubP08fHXv6wuXrzP26b8360dHucZvud9OP93s3eQn96GAfuOCCP_ye9x7EHbm3LOeG1msje0K-sMbTnuhjN_PdnCHpJTVL6e3GLWjs7KJu04SjmqsTmnxoqqIJUtOUKoH_wuIkHJ2Go5NwfBqOT8K9A3i-e_Q7AAD__5gYlYA=
--- done: testdata/inverted_filter_geospatial_dist with config 5node: 16 tests, 1 failures
    logic.go:3370: 
        testdata/inverted_filter_geospatial_dist:262: error while processing
    logic.go:3370: testdata/inverted_filter_geospatial_dist:262: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (3.86s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4161348&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4161348&tab=artifacts#/) on master @ [506412ffdb6d0f6d4952d2599b81cb9294a123a1](https://github.com/cockroachdb/cockroach/commits/506412ffdb6d0f6d4952d2599b81cb9294a123a1):


```
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:238: EXPLAIN (DISTSQL)
        SELECT k FROM geo_table WHERE geom && 'POINT(3.0 3.0)'::geometry
        expected:
            distribution: full
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk29v0z4Qx5__XoV1T7pJ3mon7Q_kR-VPBkFdW9pKgHA1hfooEa0dbAcFVX3vKAlsS6emxA9Oujt_7vw96_bgfmxBQPRxNn4RT8jF63ixXLwfX5JFNI5eLcl3cjOf3pINmjuffNki-fA2mkelvyMyZyz4v7akN5vGk-VFeM1IeM0ue0K8iaa30XL-CShoo3CS7NCB-AwcKASwopBZs0bnjC3D--pSrAoQjEKqs9yX4RWFtbEIYg8-9VsEARNzZbL-ACgo9Em6ra4dKJjcP0DOJxsEMTzQR4V5e-FlKXCOiULbZ43ycK9_VCq_S7XCAigsskQ7QfoSXkpZfFVSFpxJWbBzBq66MlwCSbQiISPGf0Pr4JRm3kVzrH-i9ahu0q1Hi7bPm8L_5qMis8RoMuKCuFI1cT6xXlQqwmdDKVnApGTsnAGCWnXFSvFH6ilMcy_IiJ-cQ9BlDu9Mqv98fXDq6zOb7hL766E1HQUnu4ddut9PP2z2ruOCjIKjZWOccfb0PB888XuisYnnhjZoPPvMJs7RZUY7_KdVZIcVBVQbrLfdmdyucWbNumpTu9OKqwIKna-zw9qJdZ0qH_gY5q1w0A4HrXDYDoet8KAdHrTC7AheHf77HQAA__-eq8eX
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk02P0zAQhu_8itFcuiuZrZ2UD_lUPrIQ1G1LWwkQrlahHkpEawfbRUFV_ztKAru0q7YkB0szk8fzvmPNFv2PFUpMPo4HL9IhXLxOp7Pp-8ElTJNB8moG3-F6MrqBJdnbkH1ZEXx4m0ySKl6D2nAePW1O6IxH6XB2EV9xiK_4ZUfKN8noJplNPiFDYzUNszV5lJ9R4Jxh4eyCvLeuSm3rH1JdouQMc1NsQpWeM1xYRyi3GPKwIpQ4qzRMKNPkuhwZagpZvqqvvZPYr8Td5kZTiQynRWa8hK7Cl0qVX7VSpeBKlfzcgY_bMkIhZEZDzMGGb-Q8zncM7Sbc-_EhWxJKsWP_7zk1P8kF0tf5KpAj1xX7xv_Wk7JwYA30hQRfuQYfMhdk7SJ-9kQpHnGlOD93IJDRbbHK_IF7hqNNkNAXR-cQtZnDO5ubP08fHXv6wuXrzP26b8360dHucZvud9OP93s3eQn96GAfuOCCP_ye9x7EHbm3LOeG1msje0K-sMbTnuhjN_PdnCHpJTVL6e3GLWjs7KJu04SjmqsTmnxoqqIJUtOUKoH_wuIkHJ2Go5NwfBqOT8K9A3i-e_Q7AAD__5gYlYA=
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist with config 5node: 16 tests, 1 failures
    logic.go:3370: 
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:262: error while processing
    logic.go:3370: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:262: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (3.88s)
=== RUN   TestExecBuild/5node
    --- FAIL: TestExecBuild/5node (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4173661&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4173661&tab=artifacts#/) on master @ [03556f98c29c65e022d2c29cb03c99d0dc1b31b0](https://github.com/cockroachdb/cockroach/commits/03556f98c29c65e022d2c29cb03c99d0dc1b31b0):


```
    logic.go:2448: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:238: EXPLAIN (DISTSQL)
        SELECT k FROM geo_table WHERE geom && 'POINT(3.0 3.0)'::geometry
        expected:
            distribution: full
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFv0zAQx9_5FNa9dJO81U5aQH4qjAyCura0lQDhagr1UaKldrAdFFT1u6MksK2dmq55sHRn_-78_8uXDbhfGQiIvkyGb-IROXsXz-azT8NzMouG0dWc3JHr6fiGrNDc-uR7huTzh2gaVfGayIKx4GWzks5kHI_mZ-ElI-ElO-8I8T4a30Tz6VegoI3CUbJGB-IbcKAQwIJCbs0SnTO2Sm_qQ7EqQTAKqc4LX6UXFJbGIogN-NRnCAJG5sLk3R5QUOiTNKuPbSmYwj9AzicrBNHf0keFeXvheSVwiolC22U75eFe_6BSfptqhSVQuDJZsdZOkDtaWwIUZnlSJboS3kpZ_lBSlpxJWbJjC1ycynAJJNGKhIwY_xOtg0M28FNsiPVvtB7VdZp5tGi7fNeL__tRmVtiNBlwQVylmjifWC9qFeGrvpQsYFIydmwBglqdilXi99RTGBdekAE_6ENwig8fTar_vYbg0GvIbbpO7J-H1nQQHOwentL93v1wt3eTF2QQ7M0f44yzp9_r3pO4I3aG85hpvZ1rHxnOKbrcaIfPmk62XVBAtcLmB-BMYZc4sWZZt2nCcc3VCYXON7v9Joh1s1Vd8DHMW-GgHQ5a4bAdDlvhXjvca4XZHrzYvvgbAAD__-uIzaE=
            
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • filter
            │ filter: geom && '010100000000000000000008400000000000000840'
            │
            └── • index join
                │ table: geo_table@geo_table_pkey
                │
                └── • inverted filter
                    │ inverted column: geom_inverted_key
                    │ num spans: 31
                    │
                    └── • scan
                          missing stats
                          table: geo_table@geom_index
                          spans: 31 spans
            ·
            Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk99v0zAQx9_5K6x72SaZ1U7KD_mpMDII6trSVgKEoynUR4mW2sF2UFDV_x0lgY12akryYOnu8vF97yvfFtyPHAREn2bjV_GEnL-JF8vFh_EFWUTj6GpJ7sj1fHpD1mhuffo1R_LxXTSP6nhDZMlY8Lw9ydlsGk-W5-ElI-EluzgT4m00vYmW889AQRuFk3SDDsQX4JBQKKxZoXPG1qlt80OsKhCMQqaL0tfphMLKWASxBZ_5HEHAstYwx1ShHTCgoNCnWd5cey9xVIu7zbTCCihcmbzcaCfIHW1UA4VFkdaJgYTXUlbflJQVZ1JW7NQBT_syXAJJtSIhI8Z_R-sg2VEwpX8Y0fl0jSD4jv6_DbH-idajus5yjxbtgO978bceVYUlRpMRF8TVUxPnU-tFM0X44pmULGBSMnbqAIJa9cXq4Q-mpzAtvSAjftSHoI8P702m_7yG4NhrKGy2Se2vh9Z0FBztHvbpfu9-uN-7zQsyCg5WhHHG2ePv5fBRfCb29ueUacM-sufoCqMd7ok-djPbJRRQrbHdU2dKu8KZNaumTRtOG65JKHS-rfI2iHVbqgX-C_NOOOiGg0447IbDTnh4ACe7J78DAAD__0Yzm4o=
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist with config 5node: 16 tests, 1 failures
    logic.go:3379: 
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:262: error while processing
    logic.go:3379: /go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist:262: too many errors encountered, skipping the rest of the input
        --- FAIL: TestExecBuild/5node/inverted_filter_geospatial_dist (2.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/distsql_enum
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_enum
@@ -1,5 +1,4 @@
 # LogicTest: 5node-default-configs !5node-metadata
-# cluster-opt: disable-span-configs
 
 # Regression test for nested tuple enum hydration (#74189)
 statement ok
@@ -66,7 +65,7 @@ ALTER TABLE t2 INJECT STATISTICS '[
   }
 ]'
 
-query T nodeidx=1
+query T nodeidx=1 retry
 EXPLAIN (VEC)
 SELECT x from t1 WHERE EXISTS (SELECT x FROM t2 WHERE t1.x=t2.x AND t2.y='hello')
 ----

--- a/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist
+++ b/pkg/sql/opt/exec/execbuilder/testdata/inverted_filter_geospatial_dist
@@ -1,5 +1,4 @@
 # LogicTest: 5node
-# cluster-opt: disable-span-configs
 
 statement ok
 CREATE TABLE geo_table(
@@ -107,7 +106,7 @@ start_key             end_key               replicas  lease_holder
 NULL                  /1152921574000000000  {1}       1
 /1152921574000000000  NULL                  {2}       2
 
-# Distributed.
+# Distributed. TODO(treilly): This claims to be distributed, but it isn't. What gives?
 query T
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE ST_Intersects('MULTIPOINT((2.2 2.2), (3.0 3.0))'::geometry, geom) ORDER BY k
@@ -176,8 +175,9 @@ start_key             end_key               replicas  lease_holder
 NULL                  /1152921574000000000  {2}       2
 /1152921574000000000  NULL                  {2}       2
 
-# Filtering is placed at node 2.
-query T
+# Filtering is placed at node 2. We need a retry here to account for possibly
+# stale dist sender caches.
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE ST_Intersects('MULTIPOINT((2.2 2.2), (3.0 3.0))'::geometry, geom) ORDER BY k
 ----
@@ -205,6 +205,9 @@ vectorized: true
 Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlN9v2jAQx9_3V1j3UtBcsJ3QrX6iP9ItE4UuMG3VjKqM3LqoYGe2mZgq_vcppGtLK9LGD4Y7-3OX71e2b8H9noOE6NvF4CgektZpPJ6MPw_aZBwNopMJuSFnyeicXKO58umPOZKvH6MkIs5f5dqjdTjzrrV3_mUwiS9G8XDSaomOIKIj2pS0gg4jQYe123tSfohG59EkuaRlrUWbjJLTKCHHl-QGKGiT4TBdoAP5HThQEDClUFgzQ-eMLdO3m01xtgLJKOS6WPoyPaUwMxZB3oLP_RxBwtDsm6LbAwoZ-jSfb7atKZilf4CcT68R5MGaPirM6wtPSgMSTDO0XbZVHu796ZfqrnKd4QoonJj5cqGdJDeVbKAwLtIy0VVwrNTqZ6bUijOlVuylCfabMlwBSXVGAkaM_4XWwS4beBMbYv0HrcfsLJ97tGi7fNuL_-vRqrDEaNLnkrhSNXE-tV5uVATvekoxwZRi7KUJCOqsKVaKf6KewmjpJenznT6IJj58Mrm-Ow1i12kobL5I7d-H1rQvdnYPmnS_dz_Y7l3l5dMbyjgLWTXE3S9nvPpzeHR4PxgPn8UPO7fG-_BZvCcfX_S-aL_C87CJ6rGxHm033Nbc5293lu9tlX_h6UjQFUY7fNXbwdZTCphdY_U8ObO0M7ywZrZpU4WjDbdJZOh8tXpQBbGulsoPfAzzWljUw6IWDurhoBYO6-GwFu7Vw71amD2Bp-s3_wIAAP__EGcZAQ==
 
 # Filtering is at gateway node since the filter is not distributable.
+#
+# TODO(treilly): What the text above claims does not square with the figure
+# generated below.
 query T
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE ST_CoveredBy('MULTIPOINT((2.2 2.2), (3.0 3.0))'::geometry, geom) ORDER BY k
@@ -236,7 +239,22 @@ Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlF9v2jwUxu_fT
 statement ok
 SET CLUSTER SETTING sql.spatial.experimental_box2d_comparison_operators.enabled = on
 
-query T
+query TTTI colnames,rowsort
+SELECT start_key, end_key, replicas, lease_holder FROM [SHOW RANGES FROM INDEX geo_table@geom_index]
+----
+start_key             end_key               replicas  lease_holder
+NULL                  /1152921574000000000  {2}       2
+/1152921574000000000  NULL                  {2}       2
+
+query ITTTI colnames,rowsort
+SELECT range_id, start_key, end_key, replicas, lease_holder FROM [SHOW RANGES FROM TABLE geo_table]
+----
+range_id  start_key  end_key  replicas  lease_holder
+43        NULL       NULL     {2}       2
+
+# We should see a distributed execution (though need to retry to purge possibly
+# stale dist sender caches).
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE geom && 'POINT(3.0 3.0)'::geometry
 ----
@@ -260,7 +278,7 @@ vectorized: true
 ·
 Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlFFv0zAQx9_5FNa9dJO81U5aQH4qjAyCura0lQDhagr1UaKldrAdFFT1u6MksK2dmq55sHRn_-78_8uXDbhfGQiIvkyGb-IROXsXz-azT8NzMouG0dWc3JHr6fiGrNDc-uR7huTzh2gaVfGayIKx4GWzks5kHI_mZ-ElI-ElO-8I8T4a30Tz6VegoI3CUbJGB-IbcKAQwIJCbs0SnTO2Sm_qQ7EqQTAKqc4LX6UXFJbGIogN-NRnCAJG5sLk3R5QUOiTNKuPbSmYwj9AzicrBNHf0keFeXvheSVwiolC22U75eFe_6BSfptqhSVQuDJZsdZOkDtaWwIUZnlSJboS3kpZ_lBSlpxJWbJjC1ycynAJJNGKhIwY_xOtg0M28FNsiPVvtB7VdZp5tGi7fNeL__tRmVtiNBlwQVylmjifWC9qFeGrvpQsYFIydmwBglqdilXi99RTGBdekAE_6ENwig8fTar_vYbg0GvIbbpO7J-H1nQQHOwentL93v1wt3eTF2QQ7M0f44yzp9_r3pO4I3aG85hpvZ1rHxnOKbrcaIfPmk62XVBAtcLmB-BMYZc4sWZZt2nCcc3VCYXON7v9Joh1s1Vd8DHMW-GgHQ5a4bAdDlvhXjvca4XZHrzYvvgbAAD__-uIzaE=
 
-query T
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE 'POINT(3.0 3.0)'::geometry::box2d && geom
 ----
@@ -284,7 +302,7 @@ vectorized: true
 ·
 Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUk99v2jAQx9_3V1j3Qiu5wk5gm_zEaFONiUEHSEOaUZXhG4sa7Mx2pkyI_31Ksv6gHaHJg6W7y-fO3698O3C_UhAQLW_GH0YTcnY1mi_mX8bnZB6No8sFuSPXs-lnskFz6-PvKZKvH6NZRDqMM85efu97L-KOEMPpMrgiMmcseFufZcMtUNBG4STeogPxDThQCGBFIbNmjc4ZW6Z31U8jVYBgFBKd5b5MryisjUUQO_CJTxEETMyFybo9oKDQx0la_banYHL_CDkfbxBEf0-fNObNjRel8hnGCm2XHbSHB2MGpaLbRCssgMKlSfOtdoLc0Xup8ywuE10JQymLH0rKgjMpC3bqgIu2DJdAYq1IyIjxP9E6OGYDb2PDSP9G61FdJ6lHi7bLD724r0dFZonRZMAFcaVq4nxsvahUhO_6UrKAScnYqQMIatUWK8U_U09hmntBBvyoD0EbHz6ZRP97DcGx15DZZBvbP4-j6SA4Oj1sM_3B_fBwdp0XpDOcLs9CEtKQhOcd8d_tGwSv8KR3cKsTuzdDlxnt8FXLx_YrCqg2WO-3M7ld440162pMHU4rrkoodL6u9utgpOtSecGnMG-Eg2Y4aITDZjhshHvNcK8RZs_g1f7N3wAAAP__-TrK3A==
 
-query T
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE 'LINESTRING(1.0 1.0, 5.0 5.0)'::geometry ~ geom
 ----
@@ -308,7 +326,7 @@ vectorized: true
 ·
 Diagram: https://cockroachdb.github.io/distsqlplan/decode.html#eJyUlGFv2jwQx98_n8K6N7SSC3YS9Eh-xdaFLhOFLiBt04yqjNxY1GBntpkyIfbZpySjBSRCsRRHd_bPl_9fvmzA_sxBQPj5YfQmGpOrd9F0Nv04uibTcBTezsgTGcaTe7JE_eiSbzmST-_DOCSdUTQOp7M4Gt9d8S4jvMso6XdZ9Vx3hLgLJ_fhLP5C_lToCigoneI4WaEF8RU4UPBgTqEweoHWalOlN_WmKC1BMAqZKtauSs8pLLRBEBtwmcsRBIz1jS56AVBI0SVZXm_bUtBr9wJZlywRRH9L9w7m7QfPKo0xJimaHjs4Hp4tGFSKHjOVYgkUbnW-XikryBPdSZ0WSZXoSXgrZfk9lbLkTMqSnZvg5lKGSyCJSonvEe1-oLFwygZ-iQ2R-oXGYTrMcocGTY8ferFbD8vCEK3IgAtiK9XEusQ4Uavw_-9LyTwmJWPnJiCo0kuxSvyRegqTtRNkwE_64F3iwwedqX-3wTt1GwqTrRLz-6U0HXgnq_uXVH923z-s3eQF6TDOPNaM3XtvDJk_bIt5ELDjuCMOenfgvcLR4EDTmc6N0RZaWXxV67LtnAKmS2z-DlavzQIfjF7UZZpwUnN1IkXrmtV-E0SqWao-cB_mrbDXDnutsN8O-61w0A4HrTA7gufb__4GAAD__zWM0xs=
 
-query T
+query T retry
 EXPLAIN (DISTSQL)
 SELECT k FROM geo_table WHERE geom ~ 'LINESTRING(1.0 1.0, 5.0 5.0)'::geometry::box2d
 ----

